pkg/analyzer/analyzers/mailgun/mailgun.go

```diff
@@ -24,11 +26,64 @@ type Analyzer struct {
 func (Analyzer) Type() analyzerpb.AnalyzerType { return analyzerpb.AnalyzerType_Mailgun }
 
 func (a Analyzer) Analyze(_ context.Context, credInfo map[string]string) (*analyzers.AnalyzerResult, error) {
-	_, err := AnalyzePermissions(a.Cfg, credInfo["key"])
+	key, ok := credInfo["key"]
+	if !ok {
+		return nil, errors.New("key not found in credentialInfo")
+	}
+
+	info, err := AnalyzePermissions(a.Cfg, key)
 	if err != nil {
 		return nil, err
 	}
-	return nil, fmt.Errorf("not implemented")
+	return secretInfoToAnalyzerResult(info), nil
+}
+
+func secretInfoToAnalyzerResult(info *DomainsJSON) *analyzers.AnalyzerResult {
+	if info == nil {
+		return nil
+	}
+	result := analyzers.AnalyzerResult{
+		AnalyzerType:       analyzerpb.AnalyzerType_Mailgun,
+		Bindings:           make([]analyzers.Binding, len(StringToPermission)),
+		UnboundedResources: make([]analyzers.Resource, info.TotalCount),
+	}
+
+	apiResource := analyzers.Resource{
+		Name:               "Mailgun API Key",
+		FullyQualifiedName: "Mailgun API Key",
+		Type:               "api key",
+		Metadata: map[string]any{
+			"domains_count": info.TotalCount,
+		},
+	}
```

Comment:
Do you think we need to differentiate this resource somehow? Or is it okay for all keys (across all accounts) to have access to this `Mailgun API Key` resource?

Since we're adding the `domains_count` here it seems like it's specific to *this* key.